src/tools/kata-ctl/src/ops/check_ops.rs

```diff
@@ -1,42 +1,101 @@
 // Copyright (c) 2022 Intel Corporation
 //
 // SPDX-License-Identifier: Apache-2.0
-//
+//00
+
+use crate::arch::x86_64::{check,get_checks};
+
+use crate::args::{CheckArgument, CheckSubCommand, IptablesCommand, MetricsCommand};
 
-use crate::arch;
 use crate::check;
+
 use crate::ops::version;
 
-use crate::args::{CheckArgument, CheckSubCommand, IptablesCommand, MetricsCommand};
+use crate::types::*;
 
-use anyhow::Result;
+use anyhow::{anyhow, Result};
 
 const NAME: &str = "kata-ctl";
 
+// This function retrieves the cmd function passes as argument
+fn get_builtin_check_func(name: &str) -> Result<BuiltinCmdFp> {
+    let check_list = get_checks();
+
+    for check in check_list {
+        if check.name.eq(name) {
+            return Ok(check.fp);
+        }
+    }
+
+    Err(anyhow!("Invalid command: {:?}", name))
+}
+
+// This function is called from each 'kata-ctl check' argument section
+fn handle_builtin_check(check: &str, args: &str) -> Result<()> {
```

Comment:
I think it would be better to create an enum for the check name. Something like:

```rust
#[derive(Debug, Clone, strum_macros::Display, EnumString)]
enum CheckType {
  CheckCpu,
  CheckNetwork,
  // ...
};
```